models/filter/transaction_filter_test.go

```diff
@@ -0,0 +1,50 @@
+package filter
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTransactionFilter(t *testing.T) {
+	t.Parallel()
+
+	t.Run("default filter", func(t *testing.T) {
+		filter := TransactionFilter{}
+		dbConditions := filter.ToDbConditions()
+
+		assert.Equal(t, 1, len(dbConditions))
+		assert.Nil(t, dbConditions["deleted_at"])
+	})
+
+	t.Run("empty filter with include deleted", func(t *testing.T) {
+		filter := fromJSON[TransactionFilter](`{
+			"include_deleted": true
+		}`)
+		dbConditions := filter.ToDbConditions()
+
+		assert.Equal(t, 0, len(dbConditions))
+	})
+
+	t.Run("with hex", func(t *testing.T) {
+		filter := fromJSON[TransactionFilter](`{
+			"hex": "test",
+			"include_deleted": true
+		}`)
+		dbConditions := filter.ToDbConditions()
+
+		assert.Equal(t, 1, len(dbConditions))
+		assert.Equal(t, "test", dbConditions["hex"])
```

Comment:
No 😄 It's reversed logic - when include deleted is `true` there is no condition linked to this. But when is is `false` or not defined - there is a condition with `deleted_at`.

However, here in this test file I'm focusing on the `transaction`-specific fields because the generic `model` conditions are tested in `model-filter-test.go`

Reply:
Will be moved to models package when we will adjust go client to those changes